there are two different versions of the XML processing library (Stax2 / Woodstox).

Title: Docker image 1.6.0 crash: NoSuchMethodError in org.codehaus.stax2.ri.EmptyIterator
Text: Context startup failed due to BeanCreationException: Error creating bean with name 'sessionFactory'... nested exception is java.lang.NoSuchMethodError: 'org.codehaus.stax2.ri.EmptyIterator org.codehaus.stax2.ri.EmptyIterator.getInstance()'

debian 13

TECHNICAL REPORT: XML PARSER CONFLICT ON JAVA 21 / TOMCAT 9
1. PROBLEM DESCRIPTION
When building LibrePlan 1.6.0 with the localization profile enabled (-P i18n) on Java 21 using Tomcat 9, the application compiles successfully but crashes during Spring context initialization (LocalSessionFactoryBean).

The following exception is thrown in catalina.out:
"Caused by: java.lang.NoSuchMethodError: 'org.codehaus.stax2.ri.EmptyIterator org.codehaus.stax2.ri.EmptyIterator.getInstance()'"

2. ROOT CAUSE
When the "-P i18n" profile is activated, the gettext-maven-plugin (or its transitive dependencies) pulls an obsolete XML parser library, "woodstox-core-asl-4.4.1.jar", into the final web archive.

Since the project explicitly declares a modern version ("woodstox-core-6.4.0.jar"), BOTH JAR files end up inside the "/WEB-INF/lib/" directory. This classpath duplication results in Class Shadowing, causing Hibernate to invoke a method from stax2-api that does not exist in the legacy 4.4.1 release.

3. REQUIRED FIXES IN POM.XML FILES
FIX A: Update the build block in "libreplan-webapp/pom.xml"
We must wrap the war plugin inside proper "" tags, upgrade its version to 3.4.0 for Java 21 compatibility, and add the exclusion rule.

Locate the "" block near the bottom of the file and replace it exactly with this text:

    <build>
        <finalName>libreplan-webapp</finalName>
        <testResources>
            <testResource>
                <directory>src/test/resources</directory>
                <filtering>true</filtering>
            </testResource>
        </testResources>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-war-plugin</artifactId>
                <version>3.4.0</version>
                <configuration>
                    <packagingExcludes>WEB-INF/lib/woodstox-core-asl-*.jar</packagingExcludes>
                    <archive>
                        <manifest>
                            <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
                        </manifest>
                    </archive>
                </configuration>
            </plugin>
        </plugins>
    </build>
FIX B: Optimize the localization profile in "libreplan-webapp/pom.xml"
To optimize build times and stop fetching redundant transitive dependencies from unused languages, restrict the gettext plugin using the "" tag.

Find the "i18n" profile and ensure the plugin looks exactly like this:


                    <plugin>
                        <groupId>com.googlecode.gettext-commons</groupId>
                        <artifactId>gettext-maven-plugin</artifactId>
                        <configuration>
                            <targetBundle>i18n.Messages</targetBundle>
                            <poDirectory>${project.build.sourceDirectory}/../resources/i18n</poDirectory>
                            <keys>
                                <key>ru</key>
                            </keys>
                        </configuration>
                        <executions>
                            <execution>
                                <id>dist</id>
                                <phase>generate-resources</phase>
                                <goals>
                                    <goal>dist</goal>
                                </goals>
                            </execution>
                        </executions>
                    </plugin>
FIX C: Add a safety net to the Root "pom.xml"
To prevent other submodules from packing the wrong artifact, declare the old coordinate under the global "" element with a "provided" scope:


    <dependency>
            <groupId>com.fasterxml.woodstox</groupId>
            <artifactId>woodstox-core-asl</artifactId>
            <version>4.4.1</version>
            <scope>provided</scope>
     </dependency>

4. CURRENT STATUS
With these structural fixes applied, the project compiles cleanly using the standard command:
"mvn clean install -P reports,userguide,i18n -DskipTests -Dliquibase.skip=true"

The deployment directory inside Tomcat 9 ("/WEB-INF/lib/") now correctly contains ONLY ONE clean engine instance: "woodstox-core-6.4.0.jar". The Spring application context and Hibernate schemas initialize flawlessly, rendering a fully localized, stable UI.

--- a/pom.xml
+++ b/pom.xml
@@ -577,7 +577,16 @@
                 <version>2.0.1</version>
             </dependency>
 
-            <!-- CXF -->
+            <!--
+                CXF.
+                cxf-core:3.1.7 (transitive) declares the obsolete
+                org.codehaus.woodstox:woodstox-core-asl:4.4.1 as a direct dependency.
+                That jar ships its own copy of org.codehaus.stax2.* classes and
+                collides with the modern com.fasterxml.woodstox:woodstox-core,
+                causing a NoSuchMethodError at Spring/Hibernate bootstrap
+                (LibrePlan issue #2067). Exclude it from every CXF entry that
+                pulls cxf-core.
+            -->
             <dependency>
                 <groupId>org.apache.cxf</groupId>
                 <artifactId>cxf-rt-transports-http</artifactId>
@@ -599,6 +608,10 @@
                         <groupId>org.springframework</groupId>
                         <artifactId>spring-web</artifactId>
                     </exclusion>
+                    <exclusion>
+                        <groupId>org.codehaus.woodstox</groupId>
+                        <artifactId>woodstox-core-asl</artifactId>
+                    </exclusion>
                 </exclusions>
             </dependency>
 
@@ -619,6 +632,10 @@
                         <groupId>org.springframework</groupId>
                         <artifactId>spring-context</artifactId>
                     </exclusion>
+                    <exclusion>
+                        <groupId>org.codehaus.woodstox</groupId>
+                        <artifactId>woodstox-core-asl</artifactId>
+                    </exclusion>
                 </exclusions>
             </dependency>
 
@@ -626,6 +643,12 @@
                 <groupId>org.apache.cxf</groupId>
                 <artifactId>cxf-rt-rs-client</artifactId>
                 <version>3.1.7</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>org.codehaus.woodstox</groupId>
+                        <artifactId>woodstox-core-asl</artifactId>
+                    </exclusion>
+                </exclusions>
             </dependency>
 
             <!-- Jackson provider -->
@@ -854,6 +877,12 @@
                 <artifactId>maven-war-plugin</artifactId>
                 <version>2.6</version>
                 <configuration>
+                    <!--
+                        Safety net for LibrePlan #2067: keep the obsolete
+                        woodstox-core-asl jar out of WEB-INF/lib even if some
+                        future transitive dependency re-introduces it.
+                    -->
+                    <packagingExcludes>WEB-INF/lib/woodstox-core-asl-*.jar</packagingExcludes>
                     <webResources>
                         <resource>
                             <directory>src/main/webapp/META-INF</directory>

